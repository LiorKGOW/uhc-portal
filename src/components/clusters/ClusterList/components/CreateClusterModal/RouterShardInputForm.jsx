// RouterShardInputForm shows a selection of router shard schemes for installing a cluster.
// it is meant to be used in a redux-form <Field> and expects an onChange callback.

import React from 'react';
import PropTypes from 'prop-types';
import { Field } from 'redux-form';
import ReduxVerticalFormGroup from '../../../../common/ReduxVerticalFormGroup';

function RouterShardSchemeInputComboBox(props) {
  return (
    <select {...props}>
      <option value="" key="">- Please select -</option>
      <option value="internal" key="internal">Internal</option>
      <option value="internet-facing" key="internet-facing">External</option>
    </select>
  );
}

function RouterShardInputForm(props) {
  const { name, disabled } = props;
  return (
    <div className="form-inline">
      <Field
        component={ReduxVerticalFormGroup}
        name={`${name}.label`}
        label="Label"
        type="text"
        disabled={disabled}
      />
      <Field
        component={ReduxVerticalFormGroup}
        componentClass={RouterShardSchemeInputComboBox}
        name={`${name}.scheme`}
        label="Scheme"
        disabled={disabled}
      />
    </div>
  );
}

RouterShardInputForm.propTypes = {
  name: PropTypes.string.isRequired,
  disabled: PropTypes.bool.isRequired,
};

export default RouterShardInputForm;
