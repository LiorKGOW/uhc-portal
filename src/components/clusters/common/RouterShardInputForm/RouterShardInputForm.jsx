// RouterShardInputForm shows a selection of router shard schemes for installing a cluster.
// it is meant to be used in a redux-form <Field> and expects an onChange callback.

import React from 'react';
import PropTypes from 'prop-types';
import {
  Button,
  FormGroup,
  HelpBlock,
  OverlayTrigger,
  Tooltip,
} from 'patternfly-react';
import {
  Field,
  FormSection,
  clearFields,
  change,
} from 'redux-form';
import ReduxVerticalFormGroup from '../../../common/ReduxVerticalFormGroup';

function RouterShardSchemeInputComboBox(props) {
  return (
    <OverlayTrigger overlay={<Tooltip id="router-shard-scheme-constraint-tooltip">Only external routes are currently supported</Tooltip>}>
      <select {...props} readOnly>
        <option value="" key="">- Please select -</option>
        <option value="internal" key="internal">Internal</option>
        <option value="internet-facing" key="internet-facing">External</option>
      </select>
    </OverlayTrigger>
  );
}

function RouterShardInputForm(props) {
  const {
    input: { name, value },
    meta: {
      dispatch,
      error,
      form,
      touched,
    },
    ...extraProps // any extra props not specified above
  } = props;

  // Force external routes until upstream 4.x clusters support them
  const forceExternal = (event) => {
    dispatch(change(form, `${name}.scheme`, event.target.value ? 'internet-facing' : null));
  };

  const clearForm = () => {
    dispatch(change(form, `${name}.label`, null));
    dispatch(change(form, `${name}.scheme`, null));
    dispatch(clearFields(form, false, false, `${name}.label`));
    dispatch(clearFields(form, false, false, `${name}.scheme`));
  };

  return (
    <FormGroup controlId={name} validationState={touched && error ? 'error' : null}>
      <FormSection className="form-inline" name={name}>
        <Field
          component={ReduxVerticalFormGroup}
          name="label"
          label="Label"
          type="text"
          onChange={forceExternal}
          {...extraProps}
        />
        <Field
          component={ReduxVerticalFormGroup}
          componentClass={RouterShardSchemeInputComboBox}
          name="scheme"
          label="Scheme"
          {...extraProps}
        />
        { (value.label || value.scheme) && <Button name="clear-button" bsStyle="default" bsSize="xsmall" onClick={clearForm}>Clear</Button> }
        { touched && error && <HelpBlock>{ error }</HelpBlock> }
      </FormSection>
    </FormGroup>
  );
}

RouterShardInputForm.propTypes = {
  // name: PropTypes.object.isRequired,
  // props passed by redux-form
  // collection of redux-form callbacks to be destructured into an html input element
  input: PropTypes.object.isRequired,
  // redux-form metadata like error or active states
  meta: PropTypes.object.isRequired,
  // plus other props passed from the <Field> component to the control (extraProps, incl. children)â€¦
};

export default RouterShardInputForm;
