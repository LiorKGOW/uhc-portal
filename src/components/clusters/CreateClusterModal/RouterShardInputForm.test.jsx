import React from 'react';
import { shallow } from 'enzyme';

import RouterShardInputForm from './RouterShardInputForm';

describe('<RouterShardInputForm />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<RouterShardInputForm name="foo" />);
    expect(wrapper).toMatchSnapshot();
  });
});
