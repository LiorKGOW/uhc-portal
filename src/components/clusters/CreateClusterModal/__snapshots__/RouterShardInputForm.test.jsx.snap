// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RouterShardInputForm /> renders correctly 1`] = `
<div
  className="form-inline"
>
  <Field
    component={[Function]}
    label="Label"
    name="foo.label"
    type="text"
  />
  <Field
    component={[Function]}
    componentClass={[Function]}
    label="Scheme"
    name="foo.scheme"
  />
</div>
`;
